easybuild/tools/build_from_easystack.py

```diff
@@ -0,0 +1,173 @@
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Support for easybuild-ing from multiple easyconfigs based on
+information obtained from provided file (easystack) with build specifications.
+
+:author: Denis Kristak (Inuits)
+:author: Pavel Grochal (Inuits)
+"""
+
+
+import yaml
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+
+_log = fancylogger.getLogger('easystack', fname=False)
+
+
+class Easystack(object):
+    """One class instance per easystack. General options + list of all SoftwareSpecs instances"""
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    def compose_ec_names(self):
+        """Returns a list of all easyconfig names"""
+        ec_names = []
+        for sw in self.software_list:
+            ec_to_append = '%s-%s-%s-%s.eb' % (str(sw.software), str(sw.version),
+                                               str(sw.toolchain_name), str(sw.toolchain_version))
+            if ec_to_append is None:
+                continue
+            else:
+                ec_names.append(ec_to_append)
+        return ec_names
+
+    # flags applicable to all sw (i.e. robot)
+    def get_general_options(self):
+        """Returns general options (flags applicable to all sw (i.e. --robot))"""
+        general_options = {}
+        # TODO add support for general_options
+        # general_options['robot'] = self.robot
+        # general_options['easybuild_version'] = self.easybuild_version
+        return general_options
+
+
+class SoftwareSpecs(object):
+    """Contains information about every SW that should be installed"""
+    def __init__(self, software, version, toolchain, toolchain_version, toolchain_name):
+        self.software = software
+        self.version = version
+        self.toolchain = toolchain
+        self.toolchain_version = toolchain_version
+        self.toolchain_name = toolchain_name
+        self.toolchain = toolchain
+        self.versionsuffix = None
+
+    # to be implemented
+    # def get_versionsuffix(self):
+    #     return self.versionsuffix or ''
+
+
+# implement this to your own needs - to create custom yaml/json/xml parser
+class GenericSpecsParser(object):
+    """Parent of all implemented parser classes"""
+    @ staticmethod
+    def parse(filename):
+        """Parent of all implemented parser functions (i.e. JSON, XML...)"""
+        raise NotImplementedError
+
+
+class YamlSpecParser(GenericSpecsParser):
+    """YAML file parser"""
```

Comment:
Since we call `YamlSpecParser` below directly, I don't see the point of having both `GenericSpecsParser` and `YamlSpecParser`.

I doubt we'll actually look into alternate formats in the future (sticking to a single format avoids confusion), so please rework to:

```python
class EasyStackParser(object):
    """Parser for easystack files."""
    @staticmethod
    def parse(filename):
        ...